### DISCUSSION TITLE
Architecture: from conglomerate image to CI pipeline definitions

### DISCUSSION THREAD

**Discussion by sanmai-NL** [oxsecurity/megalinter#2719] [Ideas]
MegaLinter is based on container images out of many imported layers. Wouldn't it be much more efficient to define a library of conditional jobs for CI pipelines that can be included? I don't see what requirement (current feature) couldn't be implemented based on such an architecture. GitLab CI would support this.

**Comment by nvuillam** [comment]
For @oxsecurity requirements, we also release MegaLinter images with each single linter in it, nothing prevents you to define them in conditional jobs :)

But one of the goals of MegaLinter is to be the more "plug & play" possible, and a single docker image helps achieve this goal
